hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapEntrySetMessageTask.java

```diff
@@ -18,60 +18,57 @@
 
 import com.hazelcast.client.impl.protocol.ClientMessage;
 import com.hazelcast.client.impl.protocol.codec.MapEntrySetCodec;
-import com.hazelcast.client.impl.protocol.task.AbstractAllPartitionsMessageTask;
 import com.hazelcast.instance.Node;
-import com.hazelcast.map.impl.MapEntries;
-import com.hazelcast.map.impl.MapService;
-import com.hazelcast.map.impl.operation.EntrySetOperationFactory;
+import com.hazelcast.map.impl.query.QueryResultRow;
 import com.hazelcast.nio.Connection;
 import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.TruePredicate;
 import com.hazelcast.security.permission.ActionConstants;
 import com.hazelcast.security.permission.MapPermission;
-import com.hazelcast.spi.OperationFactory;
+import com.hazelcast.util.IterationType;
 
 import java.security.Permission;
-import java.util.HashSet;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
 import java.util.Map;
-import java.util.Set;
 
 public class MapEntrySetMessageTask
-        extends AbstractAllPartitionsMessageTask<MapEntrySetCodec.RequestParameters> {
+        extends AbstractMapQueryMessageTask<MapEntrySetCodec.RequestParameters> {
 
     public MapEntrySetMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
         super(clientMessage, node, connection);
     }
 
     @Override
-    protected OperationFactory createOperationFactory() {
-        return new EntrySetOperationFactory(parameters.name);
+    protected Object reduce(Collection<QueryResultRow> result) {
+        List<Map.Entry<Data, Data>> entries = new ArrayList<Map.Entry<Data, Data>>(result.size());
+        for (QueryResultRow row : result) {
+            entries.add(new AbstractMap.SimpleEntry<Data, Data>(row.getKey(), row.getValue()));
```

Comment:
Taste issue; but I try to prevent long lines. I would use a static import for SimpleEntry:

entries.add(new SimpleEntry<Data, Data>(row.getKey(), row.getValue()));

And I would probably pull out the entry:

SimpleEntry entry = new SimpleEntry<Data, Data>(row.getKey(), row.getValue());
entries.add(entry)

This way you don't need to analyze the line.